### ISSUE TITLE
Update @middy/core peerDependency to v3

### ISSUE THREAD

**Issue by marcodali** [dbartholomae/middy-middleware-jwt-auth#272]
**Is your feature request related to a problem? Please describe.**
When this npm module is added to a package.json file that is already using @middy/core with the current latest version (3.6.2) and then you want to install all your dependencies. Some error like this is raising:

```
npm install
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: profe-santi-backend@1.0.0
npm ERR! Found: @middy/core@3.6.2
npm ERR! node_modules/@middy/core
npm ERR!   @middy/core@"^3.6.2" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @middy/core@"^2.0.0" from middy-middleware-jwt-auth@5.1.0
npm ERR! node_modules/middy-middleware-jwt-auth
npm ERR!   middy-middleware-jwt-auth@"^5.1.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

** Would you be willing to help with a PR? **
- [ X] Yes, absolutely
- [ ] Yes, with some guidance
- [ ] Unfortunately no time :'-(

## Describe the solution you'd like
Update package.json section peerDependencies.

"@middy/core": "^3.6.2"
instead of
"@middy/core": "^2.0.0"

**Comment by dbartholomae** [comment]
I'll check whether the middleware is compatible with both 2 and 3 :)

**Comment by dbartholomae** [comment]
This is now available as 5.1.1
